### PR TITLE
Feature/sckan 333

### PR DESCRIPTION
**Pull request by D-GopalKrishna**
Jira link - [SCKAN-333](https://metacell.atlassian.net/browse/SCKAN-333)

About task:

1. Backend Logic: -- By default show - show human-created notes & those associated with Invalid and Exported states.
2. Add boolean filter - include_system_notes
3. update openapi spec
4. add checkbox in Frontend, and related logic + update noteService

demo video
[Screencast from 11-17-2024 03:26:57 PM.webm](https://github.com/user-attachments/assets/a2f6a0df-7ff3-41ad-a125-a18ee4143d51)

[SCKAN-333]: https://metacell.atlassian.net/browse/SCKAN-333?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: backend/composer/api/filtersets.py [hunk 1]
```diff
@@ -203,11 +204,27 @@ class NoteFilter(django_filters.FilterSet):
...
 
+    @staticmethod
+    def get_notes(queryset, name, value):
+        if value:
+            return queryset
+        system_user = User.objects.get(username="system")
+        # BELOW LOGIC: if include_system_notes is False, show - system notes associated with "invalid" or "exported" + non_system_notes
+        combined_queryset = queryset.filter(
+            Q(user=system_user, note__icontains="invalid") |
+            Q(user=system_user, note__icontains="exported") |
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply backend/composer/api/filtersets.py:223]: User experience wise is not obvious why I'm seeing system notes in a situation like this:
image

File: backend/composer/api/filtersets.py [hunk 2]
```diff
@@ -203,11 +204,27 @@ class NoteFilter(django_filters.FilterSet):
...
     class Meta:
         model = Note
         fields = []
 
+    @staticmethod
+    def get_notes(queryset, name, value):
+        if value:
+            return queryset
+        system_user = User.objects.get(username="system")
+        # BELOW LOGIC: if include_system_notes is False, show - system notes associated with "invalid" or "exported" + non_system_notes
```
Comment (afonsobspinto) [inline backend/composer/api/filtersets.py:220]: nitpik: "Below Logic" is Implied, there's no need to include that expression in the comment

File: frontend/src/apiclient/backend/api.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 
 import type { Configuration } from './configuration';
-import type { AxiosPromise, AxiosInstance, RawAxiosRequestConfig } from 'axios';
+import type { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
```
[the remark this answers is not in the document]
Comment (D-GopalKrishna) [reply frontend/src/apiclient/backend/api.ts]: missed it, it was an older one I believe, fixed it now. thanks!!

File: backend/composer/api/filtersets.py [hunk 4]
```diff
@@ -218,9 +218,10 @@ def get_notes(queryset, name, value):
             return queryset
         system_user = User.objects.get(username="system")
         combined_queryset = queryset.filter(
-            Q(user=system_user, note__icontains="invalid") |
+            (Q(user=system_user, note__icontains="invalid") |
             Q(user=system_user, note__icontains="exported") |
-            ~Q(user=system_user)
+             ~Q(user=system_user)) &
+            ~Q(type__in=[NoteType.TRANSITION])
```
Comment (D-GopalKrishna) [inline backend/composer/api/filtersets.py:224]: Do not show any transition related notes by default

**Review by ddelpiano — APPROVED** [review]

**Comment by ddelpiano** [comment]
@D-GopalKrishna an issue here - the export transition should show up as per acceptance criteria

[scrnli_XAuCM5d7WaQ1mi.webm](https://github.com/user-attachments/assets/8674a484-cae7-4cc2-a6b9-752b12440d88)

**Review by ddelpiano — CHANGES_REQUESTED** [review]
exported missing the filter

**Comment by D-GopalKrishna** [comment]
thanks @ddelpiano ! Updated it - by removing the logic to - not include the transition notes. 

https://github.com/user-attachments/assets/7f42184e-cb83-473d-82c4-61a1b00d2b10

**Review by ddelpiano — APPROVED** [review]

**Review by afonsobspinto — APPROVED** [review]

**Pull request merged**